### PR TITLE
feature(workflow) - New endpoint set statuses order

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

 New endpoint set statuses order

Issue #1791  
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/workflow/src/Application/Validator/StatusIdsContainAllValidator.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_array($value)) {
+            throw new UnexpectedValueException($value, 'array');
+        }
+
+        $allStatusIds = $this->query->getAllStatusIds();
```
Comment (rprzedzik) [inline module/workflow/src/Application/Validator/StatusIdsContainAllValidator.php:46]: there is method getDictionary in StatusQueryInterface should be enough ...

File: module/workflow/src/Application/Form/DataTransformer/StatusIdDataTransformer.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+declare(strict_types=1);
+
+namespace Ergonode\Workflow\Application\Form\DataTransformer;
+
+use Ergonode\SharedKernel\Domain\Aggregate\StatusId;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+class StatusIdDataTransformer implements DataTransformerInterface
```
Comment (rprzedzik) [inline module/workflow/src/Application/Form/DataTransformer/StatusIdDataTransformer.php:16]: IF i good remember we resignee from using DataTransfromers ... for simplest form models

File: module/workflow/src/Application/Form/Model/StatusOrderSetFormModel.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+declare(strict_types=1);
+
+namespace Ergonode\Workflow\Application\Form\Model;
+
+use Ergonode\Workflow\Application\Validator as WorkflowAssert;
+
+class StatusOrderSetFormModel
+{
+    /**
+     * @WorkflowAssert\StatusIdsContainAll()
```
Comment (rprzedzik) [inline module/workflow/src/Application/Form/Model/StatusOrderSetFormModel.php:17]: What about valdation when there is not exist status or incorrect uuid ??

File: module/workflow/src/Application/Form/Type/StatusIdsType.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+namespace Ergonode\Workflow\Application\Form\Type;
+
+use Ergonode\Workflow\Application\Form\DataTransformer\StatusIdDataTransformer;
+use Ergonode\Workflow\Domain\Provider\StatusIdsProvider;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class StatusIdsType extends AbstractType
```
Comment (rprzedzik) [inline module/workflow/src/Application/Form/Type/StatusIdsType.php:19]: is there relay need add dedicated Form Type for this data ??

File: module/workflow/src/Domain/Provider/StatusIdsProvider.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Workflow\Domain\Provider;
+
+use Ergonode\Workflow\Domain\Query\StatusQueryInterface;
+
+class StatusIdsProvider
```
Comment (rprzedzik) [inline module/workflow/src/Domain/Provider/StatusIdsProvider.php:14]: IMHO this class is redundant

**Review by rprzedzik — CHANGES_REQUESTED** [review]

**Review by rprzedzik — APPROVED** [review]

File: module/workflow/src/Application/Validator/StatusIdsContainAllValidator.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+        sort($allStatusIds);
+        sort($value);
```
Comment (piotrkreft) [inline module/workflow/src/Application/Validator/StatusIdsContainAllValidator.php:49]: Using `array_diff` would give a clearer overview of what you're trying to achieve I think

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

**Pull request merged**